components/places/ios/Places/Bookmark.swift

```diff
@@ -0,0 +1,254 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+/// Snarfed from firefox-ios, although we don't have the fake desktop root,
+/// and we only have the `All` Set.
+public struct BookmarkRoots {
+
+    public static let RootGUID =          "root________"
+    public static let MobileFolderGUID =  "mobile______"
+    public static let MenuFolderGUID =    "menu________"
+    public static let ToolbarFolderGUID = "toolbar_____"
+    public static let UnfiledFolderGUID = "unfiled_____"
+
+
+    public static let All = Set<String>([
+        BookmarkRoots.RootGUID,
+        BookmarkRoots.MobileFolderGUID,
+        BookmarkRoots.MenuFolderGUID,
+        BookmarkRoots.ToolbarFolderGUID,
+        BookmarkRoots.UnfiledFolderGUID,
+    ])
+}
+
+
+/**
+ * Enumeration of the type of a bookmark item.
+ */
+public enum BookmarkNodeType: Int32 {
+    // Note: these values need to match the Rust BookmarkType
+    // enum in types.rs
+    case bookmark = 1
+    case folder = 2
+    case separator = 3
+    // The other node types are either queries (which we handle as
+    // normal bookmarks), or have been removed from desktop, and
+    // are not supported
+}
+
+
+/**
+ * A base class containing the set of fields common to all nodes
+ * in the bookmark tree.
+ */
+public class BookmarkNode {
```

Comment:
Many of the kotlin comments above apply here too - including "local timestamp" and "it's" ;)